Resources/doc/4-integrating_fosub.md

```diff
@@ -0,0 +1,188 @@
+Step 4: Integrating with FOSUserBundle
+======================================
+
+If you use FOSUserBundle and you want to integrate it with HWIOAuthBundle this article will describe most mahor
+FOSUserBundle is the most popular Symfony2 bundle, and when you use HWIOAuthBundle for allow to users to authenticate
+through different networks and services you obviously want to integrate you User class with HWIOauthBundle.
+
+Consider that you have been successfully configure any provider (e.g [Adding "Facebook Connect" functionality](bonus/facebook-connect.md))
+
+### 1) Add resource properties for User class
+
+Update your User class, by add properties for storing provider user id and access token. It may looks like:
+
+```php
+namespace MyBundle\Entity;
+
+use FOS\UserBundle\Model\User as FOSUBUser;
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * @ORM\Table(name="users")
+ * @ORM\Entity
+ */
+class User extends FOSUBUser
+{
+    /**
+     * @ORM\Column(name="id", type="integer")
+     * @ORM\Id
+     * @ORM\GeneratedValue(strategy="AUTO")
+     */
+    protected $id;
+
+    /**
+     * @ORM\Column(name="facebook_id", type="string", length=255, nullable=true)
+     */
+    private $facebookId;
+
+    /**
+     * @ORM\Column(name="facebook_access_token", type="string", length=255, nullable=true)
+     */
+    private $facebookAccessToken;
+
+    /**
+     * @return integer
+     */
+    public function getId()
+    {
+        return $this->id;
+    }
+
+    /**
+     * @param string $facebookId
+     * @return User
+     */
+    public function setFacebookId($facebookId)
+    {
+        $this->facebookId = $facebookId;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFacebookId()
+    {
+        return $this->facebookId;
+    }
+
+    /**
+     * @param string $facebookAccessToken
+     * @return User
+     */
+    public function setFacebookAccessToken($facebookAccessToken)
+    {
+        $this->facebookAccessToken = $facebookAccessToken;
+
+        return $this;
+    }
+
+    /**
+     * @return string
+     */
+    public function getFacebookAccessToken()
+    {
+        return $this->facebookAccessToken;
+    }
+```
+After adding extra properties to User entity, you need to extend base FOSUBUserProvider (if you want to add more advanced behavior, than provided from the box)
+
+
+### 2) Extend default FOSUBUserProvider
+
+The bundle provide bridge class for connect FOSUserBundle User class and HWIOAuth out of the box.
+You should extend it if you want to add more advanced behavior.
```

Comment:
What is the behavior that I get "out of the box"? I need to know that to know if I need to extend it.